src/pe/exception.rs

```diff
@@ -0,0 +1,810 @@
+//!     there can be no exception handler associated with this exception for this function, and the
+//!     effects of the prolog must be undone to compute the context of the caller function. The RIP
+//!     is within the prolog if the distance from the function start to the RIP is less than or
+//!     equal to the prolog size encoded in the unwind info. The effects of the prolog are unwound
+//!     by scanning forward through the unwind codes array for the first entry with an offset less
+//!     than or equal to the offset of the RIP from the function start, then undoing the effect of
+//!     all remaining items in the unwind code array. Step 1 is then repeated.
+//!
+//!  3. If the RIP is not within a prolog or epilog and the function has an exception handler, then
+//!     the language-specific handler is called. The handler scans its data and calls filter
+//!     functions as appropriate. The language-specific handler can return that the exception was
+//!     handled or that the search is to be continued. It can also initiate an unwind directly.
+//!
+//! For more information, see [x64 exception handling].
+//!
+//! [`ExceptionData`]: struct.ExceptionData.html
+//! [`RuntimeFunction`]: struct.RuntimeFunction.html
+//! [`UnwindInfo`]: struct.UnwindInfo.html
+//! [x64 exception handling]: https://docs.microsoft.com/en-us/cpp/build/exception-handling-x64?view=vs-2017
+
+use core::cmp::Ordering;
+use core::fmt;
+
+use scroll::ctx::TryFromCtx;
+use scroll::{self, Pread};
+
+use error;
+
+use pe::data_directories;
+use pe::section_table;
+use pe::utils;
+
+/// The function has an exception handler that should be called when looking for functions that need
+/// to examine exceptions.
+const UNW_FLAG_EHANDLER: u8 = 0x01;
+/// The function has a termination handler that should be called when unwinding an exception.
+const UNW_FLAG_UHANDLER: u8 = 0x02;
+/// This unwind info structure is not the primary one for the procedure. Instead, the chained unwind
+/// info entry is the contents of a previous `RUNTIME_FUNCTION` entry. If this flag is set, then the
+/// `UNW_FLAG_EHANDLER` and `UNW_FLAG_UHANDLER` flags must be cleared. Also, the frame register and
+/// fixed-stack allocation fields must have the same values as in the primary unwind info.
+const UNW_FLAG_CHAININFO: u8 = 0x04;
+
+/// info == register number
+const UWOP_PUSH_NONVOL: u8 = 0;
+/// no info, alloc size in next 2 slots
+const UWOP_ALLOC_LARGE: u8 = 1;
+/// info == size of allocation / 8 - 1
+const UWOP_ALLOC_SMALL: u8 = 2;
+/// no info, FP = RSP + UNWIND_INFO.FPRegOffset*16
+const UWOP_SET_FPREG: u8 = 3;
+/// info == register number, offset in next slot
+const UWOP_SAVE_NONVOL: u8 = 4;
+/// info == register number, offset in next 2 slots
+const UWOP_SAVE_NONVOL_FAR: u8 = 5;
+/// info == XMM reg number, offset in next slot
+const UWOP_EPILOG: u8 = 6;
+/// info == XMM reg number, offset in next 2 slots
+const UWOP_SPARE_CODE: u8 = 7;
```

Comment:
These are confusing due to the version 1/2 differences and the mismatch between comments and constant names.

Reply:
Funny enough, the original codes were deprecated and removed even in version one, and apparently not emitted anymore. I've updated the comments, that was oversight when copying stuff around.